### PR TITLE
Install CMake modules into ESMF_INSTALL_CMAKEDIR

### PR DESCRIPTION
**Pull request by danrosen25**
* add how-to use documentation to FindESMF.cmake
* copy *.cmake files into ESMF_INSTALL_CMAKEDIR (default ESMF_INSTALL_PREFIX/cmake)
* update ESMF user's guide
* add README.md in cmake directory (do not copy into installation as this might override README.md fill in a user's folder)

Resolves: https://github.com/esmf-org/esmf/issues/279

### CODE REVIEW

**Review by billsacks — APPROVED** [review]
Looks great!

File: src/doc/ESMF_install.tex [hunk 1]
```diff
@@ -1322,6 +1329,7 @@ \subsection{Installing the ESMF}
 \item {\tt ESMF\_INSTALL\_MODDIR} -- where to install Fortran module files.
 \item {\tt ESMF\_INSTALL\_BINDIR} -- where to install application files.
 \item {\tt ESMF\_INSTALL\_DOCDIR} -- where to install documentation files.
+\item {\tt ESMF\_INSTALL\_CMAKEDIR} -- where to install cmake module files.
```
Comment (theurich) [inline src/doc/ESMF_install.tex:1332]: None of the other items have a period (.) at the end of sentence. Maybe all should get one? Either way fine with me, just should be consistent.

File: cmake/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# CMake Modules Provided by ESMF
+
+The ESMF team provides CMake[1] modules that are useful when ESMF is needed
+within a CMake based build system. These modules can find libraries, create
+targets, and set variables within the scope of a CMake build. They may be
+used as is or customized for application specific build systems.
```
Comment (theurich) [inline cmake/README.md:6]: Could we add a sentence at the end of this paragraph to point out that CMake module files are typically kept with the project that use them? Maybe:
"Typically CMake module files are copied into the project that use them. They become part of the project and are maintained with it."

**Review by theurich — CHANGES_REQUESTED** [review]
Looks very good. Just a couple of small suggestions.

File: src/doc/ESMF_install.tex (hunk 1 above)
Comment (theurich) [reply src/doc/ESMF_install.tex:1332]: I could have sworn that there was no period when I reviewed this... but all I can say is that I definitely need (better) glasses! This looks good as is!

**Review by theurich — APPROVED** [review]
Looks good to me now. Thanks for adding the extra sentences to the usage discussion.

We should discuss during the core call whether this could go into 8.7.0, or should wait for 8.8.0.

**Comment by danrosen25** [comment]
Bring into 8.7.0 after the ESMX Fix

**Pull request merged**